src/stan/error_handling/matrix/check_cov_matrix.hpp

```diff
@@ -16,22 +16,26 @@ namespace stan {
      * covariance matrix.  A valid covariance matrix must be square,
      * symmetric, and positive definite.
      *
-     * @param function
-     * @param y Matrix to test.
-     * @param name
-     * @return <code>true</code> if the matrix is a valid covariance matrix.
-     * @return throws if any element in matrix is nan
      * @tparam T Type of scalar.
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix to test
+     *
+     * @return <code>true</code> if the matrix is a valid covariance matrix
+     * @throw <code>std::invalid_argument</code> if the matrix is not square
+     *   or if the matrix is 0x0
+     * @throw <code>std::domain_error</code> if the matrix is not symmetric, 
+     *   if the matrix is not positive definite, 
+     *   or if any element of the matrix is nan
      */
     // FIXME: update warnings
     template <typename T_y>
     inline bool check_cov_matrix(const std::string& function,
                                  const std::string& name,
                                  const Eigen::Matrix<T_y,Eigen::Dynamic,Eigen::Dynamic>& y) {
-      check_size_match(function, 
-                       "Rows of covariance matrix", y.rows(),
-                       "columns of covariance matrix", y.cols());
-      check_positive(function, "rows", y.rows());
+      check_square(function, name, y);
+      check_positive_index(function, name, "rows", y.rows());
       check_symmetric(function, name, y);
       check_pos_definite(function, name, y);
```

Comment:
Shouldn't `check_pos_definite` entail all these other checks?  

In most contexts, we'll be doing something else that checks positive definiteness and don't want to call this directly anyway.